### PR TITLE
adding interface tests

### LINKED ISSUE #53: Addition of interface tests
The test should have these scenarios
* configure two interface
* configure flow from one interface to another
* check for mac resolution
* start traffic
* verify frames rx and tx

### PR DESCRIPTION
**Pull request by Vibaswan**
Fixes #53

### CODE REVIEW

**Comment by Vibaswan** [comment]
https://github.com/open-traffic-generator/conformance/actions/runs/4026075109/jobs/6920335189


sometimes CI failing not sure whats causing the actual issue any Idea ?

**Comment by Vibaswan** [comment]
fixed it was an issue with nil check

File: helpers/otg/states.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	sr := api.NewStatesRequest()
+	sr.Ipv4Neighbors()
+	res, err := api.GetStates(sr)
+	o.LogWrnErr(nil, err, true)
+
+	tb := table.NewTable(
+		"IPv4 State Info",
+		[]string{
+			"Ethernet Name",
+			"IPv4 Addess",
```
Comment (ashutshkumr) [inline helpers/otg/states.go:26]: *Address

File: features/devices/ipv4_fowarding_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		"rxPrefix":  int32(24),
+	}
+
+	api := otg.NewOtgApi(t)
+	c := interfaceConfig(api, testConst)
+
+	api.SetConfig(c)
+
+	api.WaitFor(
+		func() bool { return macResolutionOk(api) },
```
Comment (ashutshkumr) [inline features/devices/ipv4_fowarding_test.go:34]: macResolutionOk -> ipv4NeighborsOk

File: features/devices/ipv4_fowarding_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+	api.StartTransmit()
+
+	api.WaitFor(
+		func() bool { return flowMetricsOk(api, testConst) },
+		&otg.WaitForOpts{FnName: "WaitForFlowMetrics"},
+	)
+}
+
+func interfaceConfig(api *otg.OtgApi, tc map[string]interface{}) gosnappi.Config {
```
Comment (ashutshkumr) [inline features/devices/ipv4_fowarding_test.go:46]: interfaceConfig -> ipv4ForwardingConfig

File: features/devices/ipv4_fowarding_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+			return false
+		}
+
+	}
+
+	return true
+}
+
+func macResolutionOk(api *otg.OtgApi) bool {
+	stateInfo := api.GetIpv4State()
```
Comment (ashutshkumr) [inline features/devices/ipv4_fowarding_test.go:128]: stateInfo -> neighbors

File: features/devices/ipv4_fowarding_test.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		return false
+	}
+
+	for _, s := range stateInfo {
+		if !s.HasLinkLayerAddress() {
+			return false
+		}
+	}
+
+	return true
```
Comment (ashutshkumr) [inline features/devices/ipv4_fowarding_test.go:139]: return len(neighbors) > 0

File: features/devices/ipv4_fowarding_test.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+func macResolutionOk(api *otg.OtgApi) bool {
+	stateInfo := api.GetIpv4State()
+	if stateInfo == nil {
+		return false
+	}
```
Comment (ashutshkumr) [inline features/devices/ipv4_fowarding_test.go:131]: won't be needed after suggested changes in return statement

File: features/devices/test_ipv4_forwarding.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
+import logging as log
+import pytest
+from helpers.otg import otg
+
```
Comment (ashutshkumr) [inline features/devices/test_ipv4_forwarding.py:4]: please make similar changes as in go test

**Review by ashutshkumr — CHANGES_REQUESTED** [review]
The tests need to be put under `features/devices/interfaces` dir

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**